runtime-modules/council/src/lib.rs

```diff
@@ -810,8 +804,11 @@ impl<T: Trait> EnsureChecks<T> {
         let candidate = Candidates::<T>::get(council_user_id);
 
         // prevent user from releasing candidacy stake during election
-        if candidate.cycle_id == CurrentAnnouncementCycleId::get() {
-            return Err(Error::StakeStillNeeded);
+        if candidate.cycle_id == AnnouncementPeriodNr::get() {
```

Comment:
Can you rewrite this as a direct `&&` check, where second check is `matches` macro.

Reply:
Done